packages/core/src/test/amazonqFeatureDev/prepareRepoData.test.ts

```diff
@@ -0,0 +1,110 @@
+/*!
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+import assert from 'assert'
+import { WorkspaceFolder } from 'vscode'
+import { performanceTest } from '../../shared/performance/performance'
+import { createTestWorkspace } from '../testUtil'
+import { prepareRepoData, TelemetryHelper } from '../../amazonqFeatureDev'
+import { AmazonqCreateUpload, getRandomString, Metric } from '../../shared'
+
+type resultType = {
+    zipFileBuffer: Buffer
+    zipFileChecksum: string
+}
+
+describe('prepareRepoDataPerformanceTest', function () {
+    function verifyResult(result: resultType, telemetry: TelemetryHelper, expectedSize: number): void {
+        assert.ok(result)
+        assert.strictEqual(Buffer.isBuffer(result.zipFileBuffer), true)
+        assert.strictEqual(telemetry.repositorySize, expectedSize)
+        assert.strictEqual(result.zipFileChecksum.length, 44)
+    }
+    /**
+     * Tests 250 files w/ 10 bytes each.
+     * Running more files can lead to flaky test from timeout.
```

Comment:
would there be any value on seeing how this number scales? e.g. for every 50 files it roughly increases the duration by 2 or is it too unpredictable?

Reply:
duration in general was very flaky on this one, and also it seemed to not scale linearly. I would be surprised based on what I've seen so far if an increase of 50 files corresponded so some constant duration increase. I could look into it with less files.